pkg/operator/ceph/cluster/cluster_cleanup.go

```diff
@@ -0,0 +1,145 @@
+)
+
+var (
+	appName              = "cluserCleanUp"
+	dataDirHostPath      = "ROOK_DATA_DIR_HOST_PATH"
+	cleanDataDirHostPath = "ROOK_CLEAN_DATA_DIR_HOST_PATH"
+)
+
+func (c *ClusterController) startClusterCleanUp(cluster *cephv1.CephCluster, retryInterval time.Duration) {
+	logger.Infof("starting clean up for cluster %q", cluster.Name)
+	err := c.waitForCephDaemonCleanUp(cluster, time.Duration(clusterDeleteRetryInterval)*time.Second)
+	if err != nil {
+		logger.Errorf("failed to wait till ceph daemons are destroyed. %v", err)
+		return
+	}
+	c.startCleanUpJob(cluster)
+}
+
+func (c *ClusterController) startCleanUpJob(cluster *cephv1.CephCluster) {
+	validStorage := cluster.Spec.Storage
+	privileged := true
+
+	// Get the list of all nodes in the cluster.
+	hostnameMap, err := k8sutil.GetNodeHostNames(c.context.Clientset)
+	if err != nil {
+		logger.Infof("error finding hostname. %+v", err)
+		return
+	}
+	for _, hostname := range hostnameMap {
+		storageNode := rookv1.Node{
+			Name: hostname,
+		}
+		validStorage.Nodes = append(validStorage.Nodes, storageNode)
+	}
+
+	validNodes := k8sutil.GetValidNodes(validStorage, c.context.Clientset, cephv1.GetOSDPlacement(cluster.Spec.Placement))
+
+	for i, node := range validNodes {
+		logger.Infof("starting clean up job on node %q", node.Name)
+		jobName := k8sutil.TruncateNodeName("cluster-cleanup-job-%s", strconv.Itoa(i))
+		envVars := []v1.EnvVar{
+			{Name: dataDirHostPath, Value: fmt.Sprintf("%s/%s", cluster.Spec.DataDirHostPath, cluster.Namespace)},
+			{Name: cleanDataDirHostPath, Value: "true"},
+		}
+		volumes := []v1.Volume{}
+		volumeMounts := []v1.VolumeMount{}
+		myVolume := v1.Volume{Name: "cleanup-volume", VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: cluster.Spec.DataDirHostPath}}}
+		volumes = append(volumes, myVolume)
+		myVolumeMount := v1.VolumeMount{Name: "cleanup-volume", MountPath: cluster.Spec.DataDirHostPath}
+		volumeMounts = append(volumeMounts, myVolumeMount)
+		podSpec := v1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: jobName,
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{{
+					Name:  jobName,
+					Image: c.rookImage,
+					SecurityContext: &v1.SecurityContext{
+						Privileged: &privileged,
+					},
+					VolumeMounts: volumeMounts,
+					Env:          envVars,
+					Args:         []string{"ceph", "clean"}}},
+
+				Volumes:       volumes,
+				RestartPolicy: v1.RestartPolicyNever,
+			},
+		}
+		podSpec.Spec.NodeSelector = map[string]string{v1.LabelHostname: node.Name}
+		job := &batch.Job{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      jobName,
+				Namespace: cluster.Namespace,
+				Labels:    controller.AppLabels(appName, cluster.Namespace),
+			},
+			Spec: batch.JobSpec{
+				Template: podSpec,
+			},
+		}
+
+		_, err := c.context.Clientset.BatchV1().Jobs(cluster.Namespace).Create(job)
+		if err != nil {
+			logger.Error("error running cluster clean up job on node %q. %+v", node.Name, err)
```

Comment:
```suggestion
			logger.Error("failed to run cluster clean up job on node %q. %v", node.Name, err)
```

Rook uses "failed to..." as a convention for reporting errors.